src/test/kotlin/org/rust/cargo/runconfig/RustFmtActionTestCase.kt

```diff
@@ -0,0 +1,20 @@
+package org.rust.cargo.runconfig
+
+import org.assertj.core.api.Assertions.assertThat
+import org.rust.cargo.CargoTestCaseBase
+
+class RustFmtActionTestCase : CargoTestCaseBase() {
+    override val testDataPath = "src/test/resources/org/rust/cargo/runconfig/fixtures/fmt"
+
+    fun testRustFmtFile() {
+        val cmd = "fmt"
+        val args = "-v -- --write-mode=overwrite --skip-children ./src/main.rs"
+
+        val configuration = createConfiguration(cmd, args)
+        val result = execute(configuration)
+        val filePath = "src/main.rs"
+
+        val expect = "[Bin] \"${myProject.basePath}/$filePath\"\nrustfmt --write-mode=overwrite --skip-children ./$filePath ${myProject.basePath}/$filePath\n"
+        assertThat(result.stdout == expect || result.stdout.isEmpty()).isTrue()
+    }
```

Comment:
Formatting code should be completely orthogonal to run configurations, so it is not a good idea to use RunConfigurations to test `rustfmt`. I think you can bypass the majority of the platform API here, and just test Cargo command directly, similarly to [cargo metadata](https://github.com/intellij-rust/intellij-rust/blob/master/src/test/kotlin/org/rust/cargo/commands/CargoTestCase.kt#L10)